kornia/augmentation/_2d/intensity/planckian_jitter.py

```diff
@@ -0,0 +1,134 @@
+                    [0.3757, 0.4716, 0.6171],
+                    [0.3681, 0.4717, 0.6281],
+                    [0.3609, 0.4718, 0.6380],
+                    [0.3543, 0.4719, 0.6472],
+                    [0.3480, 0.4719, 0.6555],
+                    [0.3421, 0.4719, 0.6631],
+                    [0.3365, 0.4719, 0.6702],
+                    [0.3313, 0.4719, 0.6766],
+                    [0.3263, 0.4719, 0.6826],
+                    [0.3217, 0.4719, 0.6882],
+                ]
+            )
+        else:
+            raise ValueError(
+                'Mode "' + mode + '" not supported. Please choose between "blackbody" or "CIED".')
+
+    def __getitem__(self, idx):
+        return self.pl[idx]
+
+
+class PlanckianJitter(IntensityAugmentationBase2D):
+    r"""Apply planckian jitter transformation to input tensor
+    This is physics based color augmentation, that creates realistic
+    variations in chromaticity, this can simulate the illumination
+    changes in the scene. This is based on https://arxiv.org/pdf/2202.07993.pdf
+
+    Args:
+        mode: 'blackbody' or 'CIED'.
+        same_on_batch: apply the same transformation across the batch.
+        p: probability that the random erasing operation will be performed.
+        keepdim: whether to keep the output shape the same as input (True) or broadcast it
+                                to the batch form (False).
+    Shape:
+    - Input: :math:`(C, H, W)` or :math:`(B, C, H, W)`
+    - Output: :math:`(B, C, H, W)`
+
+    Note:
+        Input tensor must be float and normalized into [0, 1]
+
+    Examples:
+        >>> rng = torch.manual_seed(0)
+        >>> input = torch.randn(1, 3, 32, 32)
+        >>> aug = PlanckianJitter(mode='CIED', p=1.0)
+        >>> aug(input)
+    """
+
+    def __init__(self,
+                 mode: str = "blackbody",
+                 same_on_batch: bool = False,
+                 p: float = 0.5,
+                 keepdim: bool = False,
+                 return_transform: Optional[bool] = None) -> None:
+        super().__init__(p=p, return_transform=return_transform, same_on_batch=same_on_batch, keepdim=keepdim)
+        self.sampler = PlanckianJitterSampler(mode=mode)  # implement the logic
+        self.idx = None
+
+    def generate_parameters(self, shape: torch.Size) -> Dict[str, Tensor]:
+        pl = self.sampler.pl
+        if self.idx is not None:
+            idx = self.idx
+        else:
+            idx = torch.randint(0, pl.shape[0], (1,)).item()
+
+        return dict(idx=idx)
+
+    def apply_transform(
+        self, input: Tensor, params: Dict[str, Tensor], transform: Optional[Tensor] = None
+    ) -> Tensor:
+
+        assert input.shape[1] > 2, 'input image should be color image'
+        idx = params['idx']
+        curr_pl = self.sampler[idx]
+        input[:, 0, ...] = input[:, 0, ...] * (curr_pl[0] / curr_pl[1])
+        input[:, 2, ...] = input[:, 2, ...] * (curr_pl[2] / curr_pl[1])
+        input[input > 1] = 1
```

Comment:
use torch.where instead. inplace function is unsafe.

Reply:
modified to use `torch.where`